File: ympa_fsharp_giraffe_server/src/model/PickupAddressDTO.fs
namespace ympa_fsharp_giraffe_server.Model

open System
open System.Collections.Generic

module PickupAddressDTO =

  //#region PickupAddressDTO


  type PickupAddressDTO = {
    Country : string;
    City : string;
    Street : string;
    House : string;
    Postcode : string;
  }
  //#endregion

File: src/main/scala/org/openapitools/models/PickupAddressDTO.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import ympa_scala_finch_server._

/**
 * Адрес доставки.
 * @param country Страна.
 * @param city Город.
 * @param street Улица.
 * @param house Номер дома.
 * @param postcode Почтовый индекс.
 */
case class PickupAddressDTO(country: Option[String],
                city: Option[String],
                street: Option[String],
                house: Option[String],
                postcode: Option[String]
                )

object PickupAddressDTO {
    /**
     * Creates the codec for converting PickupAddressDTO from and to JSON.
     */
    implicit val decoder: Decoder[PickupAddressDTO] = deriveDecoder
    implicit val encoder: ObjectEncoder[PickupAddressDTO] = deriveEncoder
}
